src/wayland/compositor/handlers.rs

```diff
@@ -294,13 +297,24 @@ where
     }
 
     fn destroyed(
-        _state: &mut D,
+        state: &mut D,
         _client_id: wayland_server::backend::ClientId,
         object_id: wayland_server::backend::ObjectId,
         data: &SurfaceUserData,
     ) {
+        let surface = state
+            .compositor_state()
+            .surfaces
+            .iter()
+            .find(|surface| surface.id() == object_id)
+            .cloned()
+            .expect("Compositor tracks creation and destruction of wl_surfaces");
```

Comment:
And when it is removed from the `surfaces` array?

Reply:
Oops, yeah I should fix that.